### PR TITLE
#538 remove autobind, refactor functions to arrow notation methods

### LINKED ISSUE #538: undefined is not an object (evaluating 'this.props.currentTitle')
## Error in mSupply Mobile

**TypeError** in **index.android.bundle:338**
undefined is not an object (evaluating 'this.props.currentTitle')

[View on Bugsnag](https://app.bugsnag.com/sustainable-solutions-nz-ltd/msupply-mobile/errors/599e622e49b8e0001869e589?event_id=59dc2fd51b0eb0001c7fe47e&i=gh&m=ci)

## Stacktrace

    index.android.bundle:338 - value

[View full stacktrace](https://app.bugsnag.com/sustainable-solutions-nz-ltd/msupply-mobile/errors/599e622e49b8e0001869e589?event_id=59dc2fd51b0eb0001c7fe47e&i=gh&m=ci)

*Created by Sustainable Solutions via Bugsnag*

### PR DESCRIPTION
**Pull request by andreievg**
Fixes #538

### CODE REVIEW

File: src/pages/CustomerInvoicePage.js [hunk 1]
```diff
@@ -274,95 +258,91 @@ export class CustomerInvoicePage extends GenericPage {
...
-            alignText: 'center',
-          },
-        ]}
-        dataTypesSynchronised={DATA_TYPES_SYNCHRONISED}
-        finalisableDataType={'Transaction'}
-        database={this.props.database}
-        selection={this.state.selection}
-        {...this.props.genericTablePageStyles}
-        topRoute={this.props.topRoute}
+  render = () => (
```
Comment (Chris-Petty) [inline src/pages/CustomerInvoicePage.js]: Pretty sure this is unnecessary. React.Component lifecycle and render functions don't need any manual binding anyway.

Though I don't see how this'll hurt - unless class variable syntax doesn't play well with inherited methods from React.Component.

File: src/pages/expansions/StocktakeEditExpansion.js [hunk 2]
```diff
@@ -122,11 +118,7 @@ export class StocktakeEditExpansion extends React.Component {
     );
   }
 
-  renderFooter() {
-    return null;
```
Comment (Chris-Petty) [inline src/pages/expansions/StocktakeEditExpansion.js]: Huh.

File: src/sync/Synchroniser.js [hunk 3]
```diff
@@ -195,7 +192,7 @@ export class Synchroniser {
    * @param  {array}   records The records to push
    * @return {Promise}         Resolves if successful, or passes up any error thrown
    */
-  async pushRecords(records) {
```
Comment (Chris-Petty) [inline src/sync/Synchroniser.js]: I think the old syntax is far more readable 😭 . In a bind I guess.

File: src/pages/CustomerInvoicePage.js (hunk 1 above)
Comment (andreievg) [reply src/pages/CustomerInvoicePage.js]: refactored

**Comment by andreievg** [comment]
Will do a little bit more testing tomorrow, not ready for merge yet

**Comment by Chris-Petty** [comment]
@andreievg How did that testing go?

**Comment by andreievg** [comment]
no extensive testing yet. 'shallow' testing went well.. will endeavour to do a full test asap

**Comment by Chris-Petty** [comment]
Probably also fixes other this.props bugs: #548 #545 #546

**Comment by andreievg** [comment]
Did a thorough test, haven't come across any issues.

**Pull request merged**